go/consensus/tendermint/apps/fuzz/fuzz.go

```diff
@@ -0,0 +1,135 @@
+	stakingApp "github.com/oasislabs/oasis-core/go/consensus/tendermint/apps/staking"
+	registry "github.com/oasislabs/oasis-core/go/registry/api"
+	//roothash "github.com/oasislabs/oasis-core/go/roothash/api"
+	staking "github.com/oasislabs/oasis-core/go/staking/api"
+)
+
+var (
+	genListGuard sync.Once
+
+	// Fuzzy is the list of all apps using the consensus interfaces and transactions.
+	Fuzzy []*App = []*App{}
+)
+
+// App describes an app on the consensus interface that has fuzzable methods.
+type App struct {
+	// Instance is an instance of the consensus application.
+	Instance abci.Application
+	// Methods is an array of fuzzable methods that this app contains.
+	Methods []transaction.MethodName
+}
+
+func GetMethodList() []*App {
+	genListGuard.Do(func() {
+		methodMap := make(map[string]*App)
+		for k, _ := range transaction.GetRegisteredMethods() {
+			appName := k.ModuleName()
+			if app, ok := methodMap[appName]; ok {
+				app.Methods = append(app.Methods, k)
+			} else {
+				methodMap[appName] = &App{
+					Methods: []transaction.MethodName{
+						k,
+					},
+				}
+			}
+		}
+
+		methodMap[epochtimemock.AppName].Instance = epochtimemock.New()
+		methodMap[registry.ModuleName].Instance = registryApp.New()
+		methodMap[staking.ModuleName].Instance = stakingApp.New()
+		Fuzzy = []*App{
+			methodMap[epochtimemock.AppName],
+			methodMap[registry.ModuleName],
+			methodMap[staking.ModuleName],
+		}
+
+		for _, v := range Fuzzy {
+			sort.Slice(v.Methods, func(i, j int) bool {
+				return v.Methods[i] < v.Methods[j]
+			})
+		}
+	})
+
+	return Fuzzy
+}
+
+// TxFunc fills and returns the transaction body for the given method with the given fuzzer.
+func TxFunc(appIndex, methodIndex int, fuzzer *gofuzz.Fuzzer) interface{} {
+	method := Fuzzy[appIndex].Methods[methodIndex]
+	bodyType := method.BodyType()
+	if bodyType == nil {
+		// Can't use the fuzzer for that.
+		return nil
+	}
+	instance := reflect.New(reflect.TypeOf(bodyType)).Interface()
+	fuzzer.Fuzz(instance)
+	return instance
+}
+
+func Fuzz(data []byte) int {
+	source := fuzz.NewRandSource(data)
+	fuzzer := gofuzz.New()
+	fuzzer = fuzzer.RandSource(source)
+
+	ctx := context.Background()
+
+	var pruneCfg abci.PruneConfig
+
+	appConfig := &abci.ApplicationConfig{
+		DataDir:         "/tmp/fuzzytest",
+		Pruning:         pruneCfg,
+		HaltEpochHeight: 1000000,
+		MinGasPrice:     1,
+	}
+
+	muxer, closer, _ := abci.NewTestMuxer(ctx, appConfig)
+	defer closer()
+	abciCtx := abci.NewContext(abci.ContextCheckTx, time.Now(), muxer.GetApplicationState())
+
+	if len(data) < 3 {
+		return -1
+	}
+
+	app := int(data[0])
+	meth := int(data[1])
+
+	if app >= len(Fuzzy) || meth >= len(Fuzzy[app].Methods) {
+		return -1
+	}
+
+	methodName := Fuzzy[app].Methods[meth]
+
+	tx := &transaction.Transaction{
+		Method: methodName,
+		Body: cbor.Marshal(TxFunc(app, meth, fuzzer)),
+	}
+
+	err := Fuzzy[app].Instance.ExecuteTx(abciCtx, tx)
```

Comment:
Hm so this doesn't mock a transaction signer for the transactions (see actual `executeTx` on the multiplexer which verifies the signature and sets the transaction signer)? Some transactions expect it being set to a valid value.

Also bypassing the multiplexer transaction processing doesn't invoke the configured `txAuthHandler`.